File: ios/TheRoyalAccommodation/ThemePreference.m
//
//  ThemePreference.m
//  ShiftHappens
//
//  Created by IT Oasis on 18/10/2025.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ThemePreference, NSObject)

RCT_EXTERN_METHOD(setDarkModeEnabled:(BOOL)enabled)
RCT_EXTERN_METHOD(getDarkModeEnabled:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/ThemePreference.swift
import Foundation
import React

@objc(ThemePreference)
class ThemePreference: NSObject {
  
  private let darkModeKey = "darkMode"

  @objc
  func setDarkModeEnabled(_ enabled: Bool) {
    UserDefaults.standard.set(enabled, forKey: darkModeKey)
  }

  @objc
  func getDarkModeEnabled(_ resolve: @escaping RCTPromiseResolveBlock,
                          rejecter reject: @escaping RCTPromiseRejectBlock) {
    let value = UserDefaults.standard.bool(forKey: darkModeKey)
    resolve(value)
  }

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
}